### DISCUSSION TITLE
Why Nebula lighthouse need a static IP?

### DISCUSSION THREAD

**Discussion by yonesmit** [slackhq/nebula#809] [Q&A] [answered]
I want to use Nebula as alternative to Tailscale, but, in the requirements I read:

"A lighthouse is the only node in a Nebula network whose IP should not change".

I want to setup lighthouse with dynamic IP (of course with a DNS that updates on each IP change). 
Really Nebula can't resolve DNS, and  later resolve from time to time to update to new DNS when it changes?
Yes, I know, there are cheap VPS, etc.  but i want to host lighthouse myself.
Is there any real technical problem that prevents resolving lighthouse DNS to check for changes?

Thanks in advance
Best Regards,

**Comment by nbrownus** [comment, marked as answer]
A DNS re-resolving PR is currently being discussed (#796) and our goal is to have support for this merged in time for the v1.7.0 release.

It would still be a best practice to run lighthouses with a static public ipv4 address since the primary purpose of a lighthouse is for overlay to underlay IP address discoverability and UDP hole punching when one or more hosts are trying to connect through a NAT.

Using DNS for lighthouse underlay IP address discovery is a choice that trades some reliability for convenience. `nebula` will not be able to query the lighthouse during the time between a dynamic IP address change, DNS record update, DNS record TTL expiry/cache flush, and finally a re-query of the DNS record. This also means DNS resolution failure is in scope for your mesh network. These trade-offs are entirely acceptable in certain networks though and in practice any issues are likely to be very minimal.

**Reply by yonesmit** [reply]
Hi nbrownus,
Thnaks a lot for the information.
That's exactly the answer I expected. Of course I know there can be small problems when lighthouse changes IP, but this (at last in my case) happens very infrequently, all I need is that the nodes refresh the lighthouse DNS frequently (as DNS TTL indicated).

Best Regards,
